lib/matplotlib/_constrained_layout.py

```diff
@@ -83,20 +85,20 @@ def do_constrained_layout(fig, renderer, h_pad, w_pad,
        A value of 0.2 for a three-column layout would have a space
        of 0.1 of the figure width between each column.
        If h/wspace < h/w_pad, then the pads are used instead.
+
+    Returns
+    -------
+    layoutgrid : private debugging structure
+        layoutgrid: useful for debugging.
     """
 
-    # list of unique gridspecs that contain child axes:
-    gss = set()
-    for ax in fig.axes:
-        if hasattr(ax, 'get_subplotspec'):
-            gs = ax.get_subplotspec().get_gridspec()
-            if gs._layoutgrid is not None:
-                gss.add(gs)
-    gss = list(gss)
-    if len(gss) == 0:
+    # make layoutgrid tree...
+    _layoutgrids = _make_layoutgrids(fig, None)
```

Comment:
`_layoutgrids` is not a dictionary (instead of being attached to the figure or gridspec) with the object as a key.

Reply:
why does the arg name have a leading underscore everywhere?

Reply:
I suspect this is meant to avoid name shadowing of `import matplotlib._layoutgrid as layoutgrid`. However, it's a bit upside down. The canonical way for preventing name collisions is *appending* an underscore. *Prepending* is associated with "private".

In this specific context, I'd change the parameter to be `layoutgrid`. And move the import to something else, `import matplotlib._layoutgrid` or `from matplotlib import _layoutgrid` or `import matplotlib._layoutgrid as mlayoutgrid`. The parameter is more prominent than the module.

Reply:
Fair enough - that was all written pretty early days for me in python, and I went overboard with underscores.  Removed most of them now.